packages/sdk-cli/src/models/localRelay/relayInfo.test.ts

```diff
@@ -72,108 +72,100 @@ describe('readRelayInfo', () => {
-//       Math.floor(timeout / pollExecTime),
-//     );
-
-//     readJsonFileSpy.mockRestore();
-//   });
-
-//   /**
-//    * First poll request's execution time 2000ms exceeds the total specified 1000ms timeout.
-//    */
-//   it('rejects on timeout', async () => {
-//     jest.useFakeTimers();
-
-//     const timeout = 1000;
-//     const readJsonFileSpy = jest.spyOn(util, 'readJsonFile').mockImplementation(
-//       () =>
-//         new Promise<false>((resolve) => setTimeout(() => resolve(false), 2000)),
-//     );
-
-//     const poll = pollRelayInfo(timeout, undefined);
-//     jest.runAllTimers();
-//     await expect(poll).rejects.toHaveProperty(
-//       'message',
-//       `Timed out after waiting for ${timeout} ms`,
-//     );
-
-//     readJsonFileSpy.mockRestore();
-//     jest.useRealTimers();
-//   });
-// });
+describe('pollRelayInfo', () => {
+  /**
+   * Poll for the relay info in interval of 100ms until either:
+   * - Valid relay info obtained
+   * - Timeout
+   *
+   * Here, we need 3 function calls to obtain valid relay info, and the interval is 100ms –
+   * more than enough time for us to poll relay info and not time out.
+   */
+  it('polls until relay info is obtained', async () => {
+    const relayInfo: RelayInfo = { port: 1, pid: 1 };
+    const readJsonFileSpy = jest
+      .spyOn(util, 'readJsonFile')
+      .mockResolvedValueOnce({})
+      .mockResolvedValueOnce({})
+      .mockResolvedValueOnce(relayInfo);
+
+    const poll = pollRelayInfo(500, 100);
+
+    await expect(poll).resolves.toEqual(relayInfo);
+
+    expect(readJsonFileSpy).toHaveBeenCalledTimes(3);
+  });
+
+  /**
+   * With timeout 500ms, and interval 200ms, pollRelayInfo can only make max. 2 poll calls + 1 initial, at t = 0.
+   * We check whether polls are really called in the specified interval and not slower/faster (so exactly 3 times).
+   * setInterval/Timeout/etc. are not guaranteed to call functions at __exactly__ the time specified,
+   * so there is a margin of 100ms (2 x 200ms + 100ms).
+   */
+  it('polls in regular intervals', async () => {
+    const timeout = 500;
+    const interval = 200;
+    const readJsonFileSpy = jest
+      .spyOn(util, 'readJsonFile')
+      .mockResolvedValue({});
+
+    await expect(pollRelayInfo(timeout, interval)).rejects.toThrow();
+
+    // Calls: 0, 1000, 2000, 3000
+    expect(readJsonFileSpy).toHaveBeenCalledTimes(
+      Math.floor(timeout / interval) + 1,
+    );
+
+    readJsonFileSpy.mockRestore();
+  });
+
+  /**
+   * With timeout 10 times larger than the interval, pollRelayInfo could be called 10 times.
+   * However, we set the first poll request's execution time at almost the timeline length – 800ms.
+   * The request is successful, so the interval should be cleared and value returned.
+   * Therefore, we check whether the interval is blocked whenever there is an existing request in-flight.
+   */
+  it('has only 1 in-flight poll request at a time', async () => {
```

Comment:
I think we had this comment previously, but this test doesn't prove this, only the total number of calls.

Also use fake timers so this test doesn't take 1 second by itself. CI as a whole takes approx 30s even on the slow GitHub actions machines so this test is ~3% of the whole runtime alone out of hundreds.